### PR TITLE
Add constant for SDM broadcast, Fixes AB#3089359

### PR DESCRIPTION
**Pull request by iamgusain**
- Add the SDM broadcast Id as constant in common
- Add log statement when sending broadcast
- Added new SDMBroadcastReceiver class for applications to register callback methods for SDM broadcasts and also to clean up msal cache when device is registered in SDM mode
 
[AB#3089359](https://identitydivision.visualstudio.com/fac9d424-53d2-45c0-91b5-ef6ba7a6bf26/_workitems/edit/3089359)

### CODE REVIEW

**Review by somalaya — APPROVED** [review]

File: common4j/src/main/com/microsoft/identity/common/java/constants/SharedDeviceModeConstants.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+class SharedDeviceModeConstants {
+    companion object {
+        /**
+         * BroadcastId of the Account Change operation.
+         */
+        const val CURRENT_ACCOUNT_CHANGED_BROADCAST_IDENTIFIER =
+            "com.microsoft.identity.client.sharedmode.CURRENT_ACCOUNT_CHANGED"
+    }
+}
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/constants/SharedDeviceModeConstants.kt]: nit: new line ending

File: common4j/src/main/com/microsoft/identity/common/java/constants/SharedDeviceModeConstants.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.microsoft.identity.common.java.constants
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/constants/SharedDeviceModeConstants.kt:24]: LICENSE

File: common4j/src/main/com/microsoft/identity/common/java/constants/SharedDeviceModeConstants.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+class SharedDeviceModeConstants {
```
Comment (shahzaibj) [inline common4j/src/main/com/microsoft/identity/common/java/constants/SharedDeviceModeConstants.kt:29]: javadoc

**Review by shahzaibj — APPROVED** [review]

File: common/src/main/java/com/microsoft/identity/common/internal/broker/SDMBroadcastReceiver.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+     * Handles the SDM broadcast and calls the callback method based on the broadcast type
+     * @param context application context.
+     * @param intent The receive intent for SDM broadcast.
+     * @param sharedDeviceModeCallback Callback to be called.
+     */
+    private static void handleSharedDeviceModeBroadCast(@NonNull final Context context,
+                                                        @NonNull final Intent intent,
+                                                        @NonNull SharedDeviceModeCallback sharedDeviceModeCallback) {
+        final String methodTag = TAG + ":handleSharedDeviceModeBroadCast";
+        final String broadcastType = intent.getStringExtra(SharedDeviceModeConstants.BROADCAST_TYPE_KEY);
```
Comment (rpdome) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/SDMBroadcastReceiver.java:90]: make sure the Intent is coming from AuthApp only?

File: common/src/main/java/com/microsoft/identity/common/internal/broker/SDMBroadcastReceiver.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        if (sSDMBroadcastReceiver == null) {
+            sSDMBroadcastReceiver = new BroadcastReceiver() {
+                @Override
+                public void onReceive(final Context context, final Intent intent) {
+                    handleSharedDeviceModeBroadCast(context, intent, sharedDeviceModeCallback);
+                }
+            };
+
+            final IntentFilter filter = new IntentFilter(SharedDeviceModeConstants.CURRENT_ACCOUNT_CHANGED_BROADCAST_IDENTIFIER);
+            context.registerReceiver(sSDMBroadcastReceiver, filter);
```
Comment (rpdome) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/SDMBroadcastReceiver.java]: do we need RECEIVER_EXPORTED flag?

File: common/src/main/java/com/microsoft/identity/common/internal/broker/SDMBroadcastReceiver.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+                .platformComponents(platformComponents)
+                .correlationId(UUID.randomUUID().toString())
+                .build();
+        return brokerMsalController.getDeviceMode(commandParameters);
+    }
+
+    /**
+     * Callback for SDM broadcasts
+     */
+    public interface SharedDeviceModeCallback {
```
Comment (rpdome) [inline common/src/main/java/com/microsoft/identity/common/internal/broker/SDMBroadcastReceiver.java:147]: i'm wondering if we should create an interface based on the actions we want app devs to do (as opposed to the steps in SDM)

e.g.

```Java
interface SDMCallBack{
    // change your UX!
    void enableSDM();

    // clear all data from the app.
    void clearData();
}
```

(in this case, clearUserData will also be invoked when we receive BROADCAST_TYPE_SDM_REGISTERED. but that would be handled in handleSharedDeviceModeBroadCast(). User doesn't need to be aware of it)

**Review by rpdome — COMMENTED** [review]
left some comments.

File: common/src/main/java/com/microsoft/identity/common/internal/broker/SDMBroadcastReceiver.java (hunk 5 above)
Comment (rpdome) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/SDMBroadcastReceiver.java]: https://developer.android.com/about/versions/14/behavior-changes-14#runtime-receivers-exported

File: common/src/main/java/com/microsoft/identity/common/internal/broker/SDMBroadcastReceiver.java (hunk 4 above)
Comment (iamgusain) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/SDMBroadcastReceiver.java:90]: the intent does not have source information in this case.

File: common/src/main/java/com/microsoft/identity/common/internal/broker/SDMBroadcastReceiver.java (hunk 6 above)
Comment (iamgusain) [reply common/src/main/java/com/microsoft/identity/common/internal/broker/SDMBroadcastReceiver.java:147]: For now, keeping this as-is to map the callbacks with individual sdm events

**Pull request merged**